### ISSUE TITLE
How to set webcam resolution?

### ISSUE THREAD

**Issue by pshinza** [abhiTronix/vidgear#2]
Thank you very much for this amazing work, it came just when I needed!

I'm trying to set the resolution to 1080p but isn't working. My code looks like this:

 # Read the video files
    options = {"hflip": True, "exposure_mode": "auto", "iso": 'auto', 
               "exposure_compensation": 'auto', "awb_mode": "horizon",
               "sensor_mode": 0} 
    
    video1 = VideoGear(enablePiCamera = False, resolution= (1920, 1080),
                       framerate=24, time_delay=2, **options).start()

**Comment by abhiTronix** [comment]
@pshinza Hello, 

> I'm trying to set the resolution to 1080p but isn't working. My code looks like this:
> Read the video files
> 
> options = {"hflip": True, "exposure_mode": "auto", "iso": 'auto', 
>            "exposure_compensation": 'auto', "awb_mode": "horizon",
>            "sensor_mode": 0} 
> 
> video1 = VideoGear(enablePiCamera = False, resolution= (1920, 1080),
>                    framerate=24, time_delay=2, **options).start() 

## Problem Insight:

As you can see in [**VideoGear Class Wiki**](https://github.com/abhiTronix/vidgear/wiki/VideoGear-Class#usage) that clearly says: 

> VideoGear Class contains enablePiCamera flag that separates access to both classes and therefore only one of them can be accessed at a time. The additional Attributes and parameters of this class are based on the class being accessed currently. 

and that **`enablePiCamera`** flag [wiki](https://github.com/abhiTronix/vidgear/wiki/VideoGear-Class#attributes-and-parameters) states:

> enablePiCamera: set this flag to access PiGear and CamGear Class respectively. This means if this flag is True, PiGear Class will be accessed and if False, the CamGear Class will be accessed. Its default value is False.

## Solution: 
 
Therefore to access [`PiGear Class and its Attributes`](https://github.com/abhiTronix/vidgear/wiki/PiGear-Class#attributes-and-parameters) in your case, you need to set the  **`enablePiCamera`** to **True** and your ** corrected algorithm will be as follow**:

```python
options = {"hflip": True, "exposure_mode": "auto", "iso": 'auto', 
           "exposure_compensation": 'auto', "awb_mode": "horizon",
           "sensor_mode": 0} 

video1 = VideoGear(enablePiCamera = True, resolution= (1920, 1080),
                   framerate=24, time_delay=2, **options).start() 
```
Hope that helps! **Also be sure to checkout the rest of the [wiki](https://github.com/abhiTronix/vidgear/wiki) .**

**Comment by pshinza** [comment]
I read the Wiki page yesterday. But since I was in doubt if there wasn't any atribbutes besides source of the CamGear class, I tried that code. 
I didn't try using PiGear because it targets Raspberry Pi Camera Modules, as you explain in the Wiki page , and I'm using two 1080p USB-webcams. 

Unfortunately, I mixed the configurations of CamGear and PiGear and only after I read the source code I understood that.

I did some changes in camgear.py to add the resolution attribute and it seems to work.

# Class CamGear with resolution attribute

```
class CamGear:
	def __init__(self, source=0, resolution=(640,480)):
		
		# initialize the camera stream and read the first frame
		self.stream = cv2.VideoCapture(source)
		self.stream.set(cv2.CAP_PROP_FRAME_WIDTH,resolution[0])
		self.stream.set(cv2.CAP_PROP_FRAME_HEIGHT,resolution[1])
		(self.grabbed, self.frame) = self.stream.read()

		# thread intialization
		self.thread=None

		# intialize termination flag
		self.terminate = False



```

**Comment by abhiTronix** [comment]
> Unfortunately, I mixed the configurations of CamGear and PiGear and only after I read the source code I understood that.

@pshinza  That's why there is a dedicated wiki section in the repo. Kindly make sure to read that instead in my advice.

>  did some changes in camgear.py to add the resolution attribute and it seems to work.
> Class CamGear with resolution attribute
> 
> class CamGear:
> 	def __init__(self, source=0, resolution=(640,480)):
> 		
> 		# initialize the camera stream and read the first frame
> 		self.stream = cv2.VideoCapture(source)
> 		self.stream.set(cv2.CAP_PROP_FRAME_WIDTH,resolution[0])
> 		self.stream.set(cv2.CAP_PROP_FRAME_HEIGHT,resolution[1])
> 		(self.grabbed, self.frame) = self.stream.read()
> 
> 		# thread intialization
> 		self.thread=None
> 
> 		# intialize termination flag
> 		self.terminate = False

`CamGear Class` **DONOT** support resolution parameter for reason, since it might break the binary if that value isn't supported by the camera being used according to my tests. I'm working on a solution and next update probably add source manuplation for`CamGear Class` also similiar to [`PiGear Class` which currently fully supports these features](https://github.com/abhiTronix/vidgear/wiki/PiGear-Class). So **stay tuned for next update** or **start a Pull Request but do take care of backward compatibity and error handling**. Goodluck.

**Comment by pshinza** [comment]
I understand and I'll stay tuned.

 Thank you very much!

**Comment by abhiTronix** [comment]
This issue will be closed when enhancement is added successfully.

**Comment by abhiTronix** [comment]
Closed in c81c2cdcd47d49ebfe30afd23324735229c1c70b
